Clipboard; export modes; .gitignore; testing

kendfss opened the pull request:
Features added:
- Copy to clipboard  
- separated export modes so that several may be chosen at once
- gitignore
- rudimentary tests via powershell
- changed default mode to ascii because it seemed more appropriate for the command line

kendfss committed:
Tweak main.py

Cancelled svg print in case of ascii.

kendfss committed:
defaults to ascii, separates html from svg, multiple file export, added clipboard support

kendfss committed:
readme tweaks

4383 commented:
Hello,

Thank you @kendfss for these changes. I'll review them ASAP.

4383 commented on README.md:
```diff
@@ -28,22 +28,31 @@ Usage examples
 Written by Hervé BERAUD
 Retrieve on Pypi => https://pypi.python.org/pypi/barcode-generator/
+
+[These Fruits, These Vegetables](https://www.youtube.com/watch?v=lHU_AJfrZlM)
```
Nice clip but I'm not sure to see the link with barcode, maybe I missed something.

kendfss replied:
It was something I did a while ago and forgot about until submitting and instantly thought "Oh no". It was about theoretically using the barcodes on stickers to identify fruits. I'll take that out.

4383 replied:
No problem. As you want.

4383 commented on README.rst:
```diff
@@ -1,5 +1,58 @@
+Written by Hervé BERAUD
+Retrieve on Pypi => https://pypi.python.org/pypi/barcode-generator/
+
+[These Fruits, These Vegetables](https://www.youtube.com/watch?v=lHU_AJfrZlM)
```
Same thing

4383 commented on README.rst:
```diff
@@ -1,5 +1,58 @@
+
+`$ barcode 0123456789123 --ascii --html --svg -o "these fruits, these vegetables"` 
+
+`$ python barcode.py`
```
I think that all the lines below should be a multi-lines code snippet to better render it.

kendfss replied:
I thought this was better because of the descriptions. I can implement the change, and it would be more consistent, but I don't think it would be more readable.

kendfss replied:
I just removed the description, in the end, the options are relatively self-explanatory. and it does look cleaner

4383 replied:
Ok no problem. Thanks

4383 commented on barcode/main.py:
```diff
@@ -94,16 +96,15 @@ def get_user_input(input_value):
+    while pred(input_value):
         try:
-            current_code = raw_input("Your code (max len 13 char): ")
+            input_value = raw_input("Your code (max length 13 char from [0, 9]): ")
```
Python 2 is dead, I don't think that we want to keep this compatibility more longer. Maybe it's the right time to clean it.

kendfss replied:
> 
> 
> Thank you for dusting off this project :)

No worries, just hope they were/will be good contributions!

4383 commented on setup.py:
```diff
@@ -67,10 +67,11 @@
         'Programming Language :: Python :: 3.2',
         'Programming Language :: Python :: 3.3',
         'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.8',
```
Might be the right time to drop the py2 support too.

May be it could be done through a follow up patch.
Thoughts?

kendfss replied:
Good idea, I'll get onto it. what should I do about the package version though?
I'm not familiar with this syntax.

4383 replied:
You just need to drop the backward compatibility by removing py27 specific syntax in the code by example by replacing `raw_input` by `input`.

When you'll substitute all these specific then you just have to drop the lines with `Python :: 2*` (2.7, 2).
After all your substitutions you should ensure that the code work as expected with py3. Maybe by using tox or virtualenv.

Let me know if I reply to your questions.

4383 reviewed (commented):
Thank you for dusting off this project :)

kendfss committed:
removed nonsensical video link; added annotation; consolidated options in readme; removed py2 support

4383 commented:
I just saw that you closed this PR, don't you want to continue with these changes?

kendfss commented:
> 
> 
> I just saw that you closed this PR, don't you want to continue with these changes?

Yes, I do!! I'm new to github/version control and assumed the protocol would be to close this request and open a new one, which I did, only to learn (3 days later) that I was on the wrong branch and had submitted a pull request to myself :smile:  
I think you should be able to see it now though  

Here's what I did
- removed nonsensical video link  
- consolidated options in readme  
- removed py2 support  
 - added annotations  
 - removed print function
 - replaced "input" with "raw_input"


PS - About versions; I meant more the version keyword argument in the setup function in setup.py file. I'm not sure how to update that. what's the versioning system?

4383 commented:
No problem :)

Thanks for your latest changes I'll review them ASAP.

Concerning the version this project is still on rc... and I didn't defined strong versioning rules like semver. So, I think we need to  release this as the first final version (without the rc), `0.1`. Let me know if that work for you.
